### PR TITLE
Add attribute executors

### PR DESCRIPTION
**Pull request by p0lunin**
This PR adds attribute executors for the `derive` attribute for the `Serialize` and `Deserialize` interfaces.

### CODE REVIEW

File: lib/lang_types.ml [hunk 1]
```diff
@@ -335,6 +337,12 @@ functor
       {branch_ty : type_; branch_var : string located; branch_stmt : stmt}
 
     and primitive = Prim of {name : string; exprs : expr list}
+
+    and attr_input = ImplInput of {impl : mk_impl; self_ty : type_}
```
Comment (cmxog) [inline lib/lang_types.ml]: Is this type meant to be extended in the future? Maybe we can indicate that (either with a comment or by making it an open type)

Also, perhaps a better name is necessary here? We're taking "input" and returning "input" below in `attr_executor`. What about `target`?

File: lib/lang_types.ml [hunk 2]
```diff
@@ -141,8 +141,10 @@ functor
             [@hash.ignore]
         mutable struct_signs : struct_sig Arena.t;
             [@hash.ignore] [@visitors.name "arena"]
-        mutable union_signs : union_sig Arena.t
-            [@hash.ignore] [@visitors.name "arena"] }
+        mutable union_signs : union_sig Arena.t;
+            [@hash.ignore] [@visitors.name "arena"]
+        attr_executors : ((string * attr_executor) list[@sexp.opaque])
+            [@visitors.opaque] [@equal.ignore] [@compare.ignore] }
```
Comment (cmxog) [inline lib/lang_types.ml:147]: Let's make this whole field `[@sexp.list]` as well so it doesn't show up in sexp dumps when there's nothing?

**Review by cmxog — COMMENTED** [review]
LGTM overall, got some minor feedback.

File: lib/lang_types.ml (hunk 2 above)
Comment (p0lunin) [reply lib/lang_types.ml:147]: This list is always non-empty, so `[@sexp.list]` will do nothing.

Comment (cmxog) [reply lib/lang_types.ml:147]: Correction (as per @p0lunin's note): there's always something, so this won't change anything.

File: lib/lang_types.ml (hunk 1 above)
Comment (p0lunin) [reply lib/lang_types.ml]: Fixed in [13ce6e0](https://github.com/tact-lang/tact/pull/84/commits/13ce6e0f81e7557b28e2c770915ba809c8f421e0)

**Pull request merged**